src/main/java/gov/cabinetoffice/gap/adminbackend/services/SchemeService.java

```diff
@@ -185,4 +185,29 @@ public void patchCreatedBy(GrantAdmin grantAdmin, Integer schemeId) {
         this.schemeRepo.save(scheme);
     }
 
+    public SchemeEntity addEditorToScheme(Integer schemeId, String editorEmailAddress, String jwt) {
+        AdminSession session = HelperUtils.getAdminSessionForAuthenticatedUser();
+        SchemeEntity scheme = this.schemeRepo.findById(schemeId).orElseThrow(EntityNotFoundException::new);
+        List<GrantAdmin> existingEditors = scheme.getGrantAdmins();
+        GrantAdmin editorToAdd;
+        try {
+            editorToAdd = userService.getGrantAdminIdFromUserServiceEmail(editorEmailAddress, jwt);
+        } catch (Exception e) {
+            throw new FieldViolationException("editorEmailAddress", "This account does not have an 'Administrator' account.");
```

Comment:
Because it's using a service method from change-grant-ownership this means quite a few different potential errors are grouped under this generic FieldViolationException, would love some advice on how to refactor this so we can still get the "Something went wrong" exception.

Reply:
Essentially, Change-grant-ownership was split across two pages which allowed us to validate the email address and confirm it in two steps. Unsure how I could maintain fieldExceptions on the front end when trying to do a double service call from there and unsure how I could maintain the the something went wrong exception when trying to do a single service call.